R/use_lintr.R

```diff
@@ -42,5 +42,32 @@ use_lintr <- function(path = ".", type = c("tidyverse", "full")) {
     )
   )
   write.dcf(the_config, config_file, width = Inf)
+
+  pkg_path <- find_package(path)
+
+  if (is.null(pkg_path)) {
+    return(invisible(config_file))
+  }
+
+  rbuildignore_path <- file.path(pkg_path, ".Rbuildignore")
+  rel_path <- xfun::relative_path(config_file, pkg_path)
```

Comment:
I tried to do this with different approaches:

- with `substr()` as in the initial PR
- with `grep()`

But tests on CI showed that this approach is too simple in many cases to be able to deal with symlinks or complex filesystems.

As mentioned in the commit message, xfun is already an indirect dependency anyways.